backend/src/controllers/goal.controller.ts

```diff
@@ -82,7 +84,18 @@ export const getGoal = async (
           res.sendStatus(httpStatus.CONFLICT).end();
         }
       }
-      res.status(httpStatus.OK).json(goalDTO).end();
+      const userEncounterIds = userCurrent?.encounters ?? [];
+      const userEncounters = await encounterService.getAllEncounters({}, userEncounterIds);
+      let progress_count = 0;
+      userEncounters.forEach((encounter) => {
+        if (encounter.date > goal.date_start && encounter.date < goal.date_end) {
```

Comment:
We need to use >= and <= comparisons for dates because a lot of times will be normalized to 00:00 so will be the same